falcon_kit/mains/get_read_ctg_map.py

```diff
@@ -0,0 +1,127 @@
+from pypeflow.data import PypeLocalFile, makePypeLocalFile, fn
+from pypeflow.task import PypeTask, PypeThreadTaskBase, PypeTaskBase
+from pypeflow.controller import PypeWorkflow, PypeMPWorkflow, PypeThreadWorkflow
+from falcon_kit.FastaReader import FastaReader
+from falcon_kit.fc_asm_graph import AsmGraph
+import argparse
+import glob
+import sys
+import subprocess as sp
+import shlex
+import os
+
+def make_dirs(d):
+    if not os.path.isdir(d):
+        os.makedirs(d)
+
+
+def get_read_ctg_map(rawread_dir, pread_dir, asm_dir):
+
+    read_map_dir = os.path.abspath(os.path.join(asm_dir, "read_maps"))
+    make_dirs(read_map_dir)
+
+    PypeMPWorkflow.setNumThreadAllowed(12, 12)
+    wf = PypeMPWorkflow()
+
+    rawread_db = makePypeLocalFile( os.path.join( rawread_dir, "raw_reads.db" ) )
+    rawread_id_file = makePypeLocalFile( os.path.join( read_map_dir, "raw_read_ids" ) )
+
+    @PypeTask( inputs = {"rawread_db": rawread_db}, 
+               outputs =  {"rawread_id_file": rawread_id_file},
+               TaskType = PypeThreadTaskBase,
+               URL = "task://localhost/dump_rawread_ids" )
+    def dump_rawread_ids(self):
+        rawread_db = fn( self.rawread_db )
+        rawread_id_file = fn( self.rawread_id_file )
+        os.system("DBshow -n %s | tr -d '>' | awk '{print $1}' > %s" % (rawread_db, rawread_id_file) )
```

Comment:
This is a recently discovered problem. On some Debian Linux systems, `awk` is really `gawk`, which is linked to `libreadline.so`, which crashes when it uses the one in smrtlink. So this must be `... | LD_LIBRARY_PATH= awk ...`, with a space after `=` to clear LD_LIBRARY_PATH. The problem only occurs for `awk`, so far as we know.

* https://bugzilla.nanofluidics.com/show_bug.cgi?id=33476